### ISSUE TITLE
Zero-weight minibatches poison PCD and UCD with NaNs

### ISSUE THREAD

**Issue by cossio** [cossio/RestrictedBoltzmannMachines.jl#143]
## Summary

Per-sample weights of zero should ignore those samples. However, if a minibatch contains only zero-weight samples, PCD first computes a normalized weighted mean by dividing by `mean(wd) == 0`. This produces `NaN`; multiplying the resulting gradient by the minibatch weight of zero does not recover it because `NaN * 0 == NaN`.

The overall weight vector can still have positive total weight, so this can occur simply because zero-weight samples are grouped in one minibatch.

Relevant code:

- [`wmean` normalization](https://github.com/cossio/RestrictedBoltzmannMachines.jl/blob/b1a7c57d/src/util/util.jl#L14-L32)
- [weighted positive phase before minibatch-weight correction](https://github.com/cossio/RestrictedBoltzmannMachines.jl/blob/b1a7c57d/src/train/pcd.jl#L75-L89)
- [the analogous UCD ordering](https://github.com/cossio/RestrictedBoltzmannMachines.jl/blob/b1a7c57d/src/train/ucd.jl#L201-L230)

## Reproduction

```julia
import RestrictedBoltzmannMachines as RBMs

data = Bool[
    0 1 0 1
    1 0 1 0
]
wts = [0.0, 0.0, 1.0, 1.0]

rbm = RBMs.BinaryRBM(
    zeros(2),
    zeros(1),
    fill(0.1, 2, 1),
)

RBMs.pcd!(
    rbm,
    data;
    wts,
    batchsize = 2,
    iters = 1,
    steps = 0,
    shuffle = false,
    rescale = false,
    zerosum = false,
)

@show all(isfinite, rbm.visible.par)
@show all(isfinite, rbm.hidden.par)
@show all(isfinite, rbm.w)
```

Observed at `b1a7c57d`:

```text
all(isfinite, rbm.visible.par) = true
all(isfinite, rbm.hidden.par) = false
all(isfinite, rbm.w) = false
```

The same ordering can affect centered and standardized PCD.

## Expected behavior

- Individual zero-weight samples contribute nothing to the objective and can be excluded when minibatches are formed.
- A minibatch with total weight zero contributes no training step and should be skipped before gradient, chain, optimizer, wrapper-statistics, and callback work.
- Skipping a zero-weight minibatch should not consume one of the requested `iters` parameter updates.
- If the complete training weight vector has total weight zero, reject it up front with a clear error because there is no data objective to estimate.

## Acceptance criteria

- The reproduction above leaves all parameters finite and still performs the requested number of updates using positive-weight samples.
- Mixed positive/zero-weight minibatches match the result obtained after removing the zero-weight samples.
- Globally all-zero weights raise a clear error before any model mutation.
- Plain, centered, and standardized PCD, plus UCD, use consistent semantics.
